content/best_practices.adoc

```diff
@@ -56,7 +56,48 @@ Some operations require logging/auditing to allow to understand what happened du
+
+* `Vue.createApp` to create a new Vue application instead of `new Vue`
+* use statement `this.$root.$applicationLoaded()` when all Data is fetched and the UI has been displayed to the End-user.
+
+=== REST API Performances
+
+By increasing the number of REST calls in a single Web-Page, this will allow to retrieve information in parallel. This was a major evolution of eXo Platform product architecture introduced in version 6 which will reduce the classic Server-End stateful DOM rendering (that was made using JSP, Servlet, Portlet, JSF, Spring MVC...). As an advantage of this, the Rendering phase is decentralized and is deported on Client-Side. Consequently, the Server will hold less information in its memory (Stateful UI Tree) and make less computing related to page rendering. As a significant drawback of this new architecture, making more parallel computing (parallel REST calls for each page) makes the multi-threading aspect more important to consider. In fact, a REST call has to be performant and must not hold any potential thread-blocking statement. In addition, the HTTP cache headers has to be considered for almost all REST calls. We have several mime types retrieved through REST calls:
+
+* CSS
+* Image
+* JSON representing a stored data
+
+For CSS and Images, it's important to consider using a https://developers.google.com/web/fundamentals/performance/get-started/httpcaching-6[long-term cache strategy]. By doing this, the Browser will always retrieve the resource from local cache (Disk or Memory cache) and will not fetch the resource from Server-End again. If the resource can be dynamic, such as illustration on news or Application Center thumbnail, you can add a suffix to the URL that references the `lastModifiedDate` property of the image (https://github.com/Meeds-io/app-center/commit/2b50274f84e9c92c05a71514bf76a802f3f60f99#diff-8a61c4f2ca478f64c19717d95c05298629a81bec5eb62dd4539528970d0a67b1R60[See example])
+
+For the JSON objects retrieved using REST API, you can consider using a https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag[ETag] to cache objects in browser and to verify that it hasn't been modified since last retrieval. This strategy is very relevant when retrieving information that is not frequently changed, such as an activity that once written, it will not change. By using this strategy, you will avoid to download content from Server-end to save Bandwidth (consequently save time) and even save computing time in Server-End (https://github.com/Meeds-io/social/blob/ffad68f249e4cbfb270e62091fe50c91eb700ecd/component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java#L345[See example]).
```

Comment:
```suggestion
For CSS and Images, it's important to consider using a https://developers.google.com/web/fundamentals/performance/get-started/httpcaching-6[long-term cache strategy]. By doing this, the Browser will always retrieve the resource from the local cache (Disk or Memory cache) and will not fetch the resource from Server-End again. If the resource can be dynamic, such as illustration on news or Application Center thumbnail, you can add a suffix to the URL that references the `lastModifiedDate` property of the image (https://github.com/Meeds-io/app-center/commit/2b50274f84e9c92c05a71514bf76a802f3f60f99#diff-8a61c4f2ca478f64c19717d95c05298629a81bec5eb62dd4539528970d0a67b1R60[See example])

For the JSON objects retrieved using REST API, you can consider using a https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag[ETag] to cache objects in browser and to verify that it hasn't been modified since the last retrieval. This strategy is very relevant when retrieving information that is not frequently changed, such as an activity that once written, will not change. By using this strategy, you will avoid downloading content from Server-end to save Bandwidth (consequently save time) and even save computing time in Server-End (https://github.com/Meeds-io/social/blob/ffad68f249e4cbfb270e62091fe50c91eb700ecd/component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java#L345[See example]).
```